tests/jobs/test_scheduler_job.py

```diff
@@ -3811,20 +3816,40 @@ def test_create_dag_runs_datasets(self, session, dag_maker):
         )
         session.add(event2)
 
+        # Create a third event, creation time is more recent, but data interval is even older
```

Comment:
Given that this is reproducible with a single, previous DAG run having a dataset event, shouldn't the test only require 3 total DAG runs?

- One DAG run from the original DAG (let's call this Run A)
- One more DAG run (Run B) after the new DAG is introduced, which triggers the new DAG's DAG run which should only have dataset info from Run B.

Reply:
The newly designed test case perfectly addresses the scenario you have mentioned. Let me know if i missing anything.